datafusion/physical-expr/src/aggregate/sum_distinct.rs

```diff
@@ -84,7 +89,12 @@ impl AggregateExpr for DistinctSum {
     }
 
     fn create_accumulator(&self) -> Result<Box<dyn Accumulator>> {
-        Ok(Box::new(DistinctSumAccumulator::try_new(&self.data_type)?))
+        macro_rules! helper {
+            ($t:ty, $dt:expr) => {
+                Ok(Box::new(DistinctSumAccumulator::<$t>::try_new(&$dt)?))
```

Comment:
I think we can do the same for `DistinctCountAccumulator`

Reply:
Quite possibly, right now I'm just doing the minimum to be able to remove the ScalarValue arithmetic kernels :sweat_smile:

Reply:
But not arithmetic :smile:

Reply:
Aha 😁 let me follow up this PR then ;)